internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -452,6 +466,7 @@ func resourceKustoClusterRead(d *pluginsdk.ResourceData, meta interface{}) error
 			if err := d.Set("optimized_auto_scale", flattenOptimizedAutoScale(props.OptimizedAutoscale)); err != nil {
 				return fmt.Errorf("setting `optimized_auto_scale`: %+v", err)
 			}
+
```

Comment:
[This comment is for line 478:] I believe if the `virtual_network_configuration` does not exist in the config file we should not be writing it to the state file, so I this should become something like this?

```suggestion
			var vnetConfig []interface{}
			if _, ok := d.GetOk("virtual_network_configuration"); ok {
				vnetConfig = flattenKustoClusterVNET(props.VirtualNetworkConfiguration)
			}
			d.Set("virtual_network_configuration", vnetConfig)
```